src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1988,6 +1995,11 @@ ReturnCode_t DataWriterImpl::check_datasharing_compatible(
     bool has_key = type_->m_isGetKeyDefined;
 
     is_datasharing_compatible = false;
+    if (is_custom_payload_pool_)
+    {
+        EPROSIMA_LOG_INFO(DATA_WRITER, "Custom payload pool detected. Data Sharing disabled.");
+        return ReturnCode_t::RETCODE_OK;
+    }
```

Comment:
I'm wondering if this couldn't be done outside of this function. check_datasharing_compatible is only used during the enable of the DataWriter, maybe we don't even need to check it if the payloadpool is a custom one.

This also applies to DataReaders but I don't see any check similar to this one being made there.